components/throttle-fire.js

```diff
@@ -0,0 +1,38 @@
+// throttle-fire.js
+
+var _ = require('underscore');
+
+var wrapper = require('../src/javascript-wrapper.js');
+
+/**
+ * This component is usually used in conjunction with the throttle component 
+ * or with the fire-throttle graph.  This component is used  to wait for 
+ * input and return the throttle size controlling the number of elements 
+ * to allow into the queue for processing. 
+ *
+ * The throttle component is usually used at the start of a pipeline to read
+ * a file of records and sends a hash with the specified number of elements 
+ * down stream to avoid flooding the request queue.  
+ * 
+ * This throttle-fire component is usually invoked at the end of a pipeline 
+ * to make the throttle fire a new set of elements for processing. 
+ * 
+ * @param throttle_size the number of elements the throttle should feed into the
+ *        pipeline now. 
+ * @param input any data input from the upstream nodes that indicates data
+ *        processing is done and the pipeline is ready for a new batch of 
+ *        data.  The content doesn't matter.  Only that it has arrived.
+ *
+ * @return the number of elements the throttle should feed into the pipeline
+ */
+module.exports = wrapper(function(throttle_size, input) {
```

Comment:
I thought about that, but didn't think so.    Ideally, I want to fill the pipeline with a lot of patient records to start with - at least one or two per VM.   That means we need to start with a throttle size of 10 or 20.  However, I don't want to continue to send 10 or 20.  I want to simply replace whatever just finished processing with another ID or two, and I wanted to make the next throttle size controllable so we could have different algorithms for what is sent next in the future. Kick just does a blind fire.  It doesn't let me send a value, AFAIK.